pkgs/build-support/coq/meta-fetch/default.nix

```diff
@@ -140,11 +151,11 @@ switch arg [
     out =
       let
         v = if isVersion arg then arg else shortVersion arg;
-        given-sha256 = release.${v}.sha256 or "";
-        sha256 = if given-sha256 == "" then fakeSha256 else given-sha256;
-        rv = release.${v} // {
-          inherit sha256;
-        };
+        givenHash = release.${v}.hash or "";
+        hash = if givenHash == "" then fakeHash else givenHash;
+        givenSha256 = release.${v}.sha256 or "";
+        sha256 = if givenSha256 == "" then fakeSha256 else givenSha256;
+        rv = release.${v} // (if release.${v} ? hash then { inherit hash; } else { inherit sha256; });
```

Comment:
I don’t understand the logic here. It seems that `{ hash = fakeHash; }` is never used and `{ sha256 = fakesha256; }` is preferred.

I’d rather see something as follows:

~~~
let
r = release.${v};
h =
if hasAttr "hash" r then
  if hasAttr "sha256" r then throw "…" else
    { hash = if r.hash == "" then fakeHash else r.hash }
else
  if r.sha256 or "" != "" then { inherit (r) sha256; } else { hash = fakeHash; };
rv = r // h;
~~~